scipy/stats/_qmc.py

```diff
@@ -898,6 +1044,63 @@ def random(self, n: IntNumber = 1) -> np.ndarray:
         self.num_generated += n
         return samples
 
+    def _random_optimal(self, n: IntNumber = 1) -> np.ndarray:
+        """Draw `n` in the half-open interval ``[0, 1)``.
+
+        Parameters
+        ----------
+        n : int, optional
+            Number of samples to generate in the parameter space. Default is 1.
+
+        Returns
+        -------
+        sample : array_like (n, d)
+            LHS sample.
+
+        """
+        if self.d == 0 or n == 0:
+            return np.empty((n, self.d))
+
+        best_sample = self._random(n=n)
+        best_disc = discrepancy(best_sample)
+
+        if n == 1:
+            return best_sample
+
+        bounds = ([0, self.d - 1],
+                  [0, n - 1],
+                  [0, n - 1])
+
+        n_nochange = 0
+        n_iters = 0
+        while "Optimization loop":
+            n_iters += 1
+
+            col = rng_integers(self.rng, *bounds[0])
+            row_1 = rng_integers(self.rng, *bounds[1])
+            row_2 = rng_integers(self.rng, *bounds[2])
+            disc = _perturb_discrepancy(best_sample,
+                                        row_1, row_2, col,
+                                        best_disc)
+            if disc < best_disc:
+                best_sample[row_1, col], best_sample[row_2, col] = (
+                    best_sample[row_2, col], best_sample[row_1, col])
+
+                best_disc = disc
+                n_nochange = 0
+            else:
+                # stopping criterion: wait for no successive improvement
+                if abs(best_disc - disc) < 1e-10:
```

Comment:
I'm not sure if `abs` is appropriate here. If `disc` is much greater than `best_disc`, it seems to me that `n_nochange` should increase. Basically, I don't think you need this additional `if` statement at all. If `disc < best_disc` above is False, then there is no change and `n_nochange` should increase. Right? Isn't the idea to just stop if no _improvement_ is made in some threshold number of iterations?

Reply:
Ok sure I will remove it. It should then be more than 10 (it's arbitrary in any case...) as the changes being totally random we might just have bad luck for a few iterations.

As you said, we will see what people want want they start to use it.

Reply:
Well maybe you had a different stopping criterion in mind, if it was intentional. You're not just looking for a local minimum but instead a local _plateau_ where it's hard to find permutations that bring the discrepancy up _or_ down?

Reply:
I have revisited this removing the abs as you suggested. I also put private attributes to control the params if someone really needs to and cannot wait a release.